### ISSUE TITLE
Weird assembly results while using ONT data

### ISSUE THREAD

**Issue by Orz-CQ** [marbl/verkko#80]
Hi,
We used hifi reads(30G) and ont(20G) reads for a plant (genome size ~500m). 
While the hifiasm's output seems to be reasonable, about 550m; the verkko's results seems to weird.
First I used the default parameter 
`verkko -d defalut_out --threads 10 --hifi hifi.fastq.gz  --nano ont.fq.gz`
the size of assembly.fasta is 586M
the BUSCO results is here
`       
        C:75.5%[S:46.6%,D:28.9%],F:1.4%,M:23.1%,n:2326
        1757    Complete BUSCOs (C)
        1085    Complete and single-copy BUSCOs (S)
        672     Complete and duplicated BUSCOs (D)
        33      Fragmented BUSCOs (F)
        536     Missing BUSCOs (M)
        2326    Total BUSCO groups searched
`
The hifi-coverage.csv 
![hifi_depth](https://user-images.githubusercontent.com/74066476/173263288-006be606-e791-4967-a95d-996554ca5c16.jpeg)
and ont-coverage.csv
![ont_depth](https://user-images.githubusercontent.com/74066476/173263395-4a411843-8a86-4c4f-a1aa-242ddd1599a8.jpeg)

I thought that there may something wrong in my ont data, I mapped my ont data to the hifiasm results.
However, the single base depth, bin size depth and summary depth (counted by mosdepth) show my ont data and hifi data indeed come from the same sample.

I tried the parameter from #77.
`verkko -d change_parameter_out --threads 10 --hifi hifi.fastq.gz  --nano .ont.fq.gz --seed-max-length 1000 --min-score 1000 --min-identity 0.75 `
the size of assembly.fasta turned to be 801M.
The hifi-coverage.csv 
![HIFI](https://user-images.githubusercontent.com/74066476/173264425-dc523a4e-29af-4eba-8521-fbbcaf3725c0.jpeg)
and ont-coverage.csv
![ont](https://user-images.githubusercontent.com/74066476/173264442-e9c76cc0-c2cb-4b8d-bc9f-6caf5c1894f8.jpeg)

Can you give me some advices?
Thank you.

Sincerely.
Lan

**Comment by skoren** [comment]
Given a 500 mb genome, I'd expect a 1gb assembly from verkko (it doesn't purge and outputs both haplotypes by default). What's your ONT coverage? What are the sizes of the GFAs in the `2-*` and `4-*` folder? It's likely this is similar to #77 and you may need to adjust the thresholds down.

**Comment by Orz-CQ** [comment]
The ONT coverage may be 40X.

The size of GFA files in 2-* are about 670M~673M in default parameter.

In default parameter, the size in 2-* GFAs

- 673M gapped-hifi-resolved.gfa
- 671M gapped-once-hifi-resolved.gfa
- 671M gapped-twice-hifi-resolved.gfa
- 670M unitig-unrolled-hifi-resolved.gfa
- 672M unrolled-hifi-resolved.gfa

the size in 4-* GFAs

- 9.4M combined-edges-uniques.gfa
- 672M connected.gfa
- 670M gapped-once-unitig-unrolled-hifi-resolved.gfa
- 670M gapped-unitig-unrolled-hifi-resolved.gfa
- 672M normal-connected.gfa
- 392M ont-resolved-graph.gfa
- 392M unitig-unrolled-ont-resolved.gfa
- 392M unrolled-ont-resolved.gfa

In this parameter,
`verkko -d change_parameter_out --threads 10 --hifi hifi.fastq.gz --nano .ont.fq.gz --seed-max-length 1000 --min-score 1000 --min-identity 0.75`
the size in 2-* GFAs

- 673M gapped-hifi-resolved.gfa
- 671M gapped-once-hifi-resolved.gfa
- 671M gapped-twice-hifi-resolved.gfa
- 670M unitig-unrolled-hifi-resolved.gfa
- 672M unrolled-hifi-resolved.gfa

the  size in 4-* GFAs
- 9.4M combined-edges-uniques.gfa
- 679M connected.gfa
- 670M gapped-once-unitig-unrolled-hifi-resolved.gfa
- 670M gapped-unitig-unrolled-hifi-resolved.gfa
- 678M normal-connected.gfa
- 586M ont-resolved-graph.gfa
- 585M unitig-unrolled-ont-resolved.gfa
- 586M unrolled-ont-resolved.gfa

**Comment by skoren** [comment]
Right, so your graph beforehand is almost 700mb in compressed space which should be >1gb post uncompression but the not-resolved graph is much smaller. It shouldn't shrink from ONT resolution unless you have pretty low ONT coverage. I'd suggest following the suggestions in #77 or you can try the low_ont_coverage branch.

**Comment by Orz-CQ** [comment]
HI.
While I tried the mentioned parameter above.
The results turned to be ummmmm.

The  assembly.fasta size is 596M,
and the BUSCO socre shows 
```
        --------------------------------------------------
        |Results from dataset eudicots_odb10              |
        --------------------------------------------------
        |C:82.2%[S:55.1%,D:27.1%],F:1.4%,M:16.4%,n:2326   |
        |1912   Complete BUSCOs (C)                       |
        |1282   Complete and single-copy BUSCOs (S)       |
        |630    Complete and duplicated BUSCOs (D)        |
        |32     Fragmented BUSCOs (F)                     |
        |382    Missing BUSCOs (M)                        |
        |2326   Total BUSCO groups searched               |
        --------------------------------------------------
```
the gfa files in 2-processGraph still shows the results
```
673M  gapped-hifi-resolved.gfa
671M  gapped-once-hifi-resolved.gfa
671M  gapped-twice-hifi-resolved.gfa
670M  uunitig-unrolled-hifi-resolved.gfa
672M  unrolled-hifi-resolved.gfa
```
Can you give me some advices?

**Comment by skoren** [comment]
To answer your previous questions 
1/2. ONT data is not corrected, it's aligned to the graph. What is the approximate identity when you align the ONT data yourself? Your ONT data may be older or lower quality than we typically have seen.
3. Yes, you'd run merqury (https://github.com/marbl/merqury), compressed -no-filt option, on the two assembly outputs from hifiasm + hic and provide the resulting markers to verkko.

The 2-processGraph files don't change when you change ONT parameters, those steps are the same. What does the 4-processONT look like now in terms of file sizes. What were the exact parameters you used for the ONT step? It seems your asm is more complete now than it was before, you may need to lower the ONT options even more (e.g. lower the 30 to 1 or 2, the 5 3 2 to 3 2 2) and use the modified alignment parameters as well.

**Comment by hrluo93** [comment]
> 

Thanks for your reply!

I meet errors when in 1-buildGraph. I had manually updated MBG by conda but it seems still have errors, could you please take a look?
[verkko.e.txt](https://github.com/marbl/verkko/files/9052803/verkko.e.txt)
[buildGraph.err.txt](https://github.com/marbl/verkko/files/9052804/buildGraph.err.txt)

Thanks a lot! 

Ran

**Comment by skoren** [comment]
There have been updates to MBG since the 1.0.11 release you're using so first thing I'd try is to download and build the tip version and try that instead to see if it fixes the error. You can just run buildGraph.sh by hand in the 1-buildGraph folder with the new MBG.

**Comment by hrluo93** [comment]
> There have been updates to MBG since the 1.0.11 release you're using so first thing I'd try is to download and build the tip version and try that instead to see if it fixes the error. You can just run buildGraph.sh by hand in the 1-buildGraph folder with the new MBG.

Thanks a lot, I will try run buildGraph.sh by hand with the 1.0.11 version, if there are still problems, I will submit a new issue tag so you can close this issue. 

Best wishes!

Ran
